plans/external.pp

```diff
@@ -0,0 +1,57 @@
+# @summary
+#   A plan that generates external facts based on the provided modulepath and
+#   sets facts on specified targets.
+#
+# @param path The path to the directory on localhost containing external facts
+# @param targets The targest the collect and set facts on
+# @return The target objects, with facts added
+# @example Gather external facts from an installed module
+#   $moduledir = module_directory('mymod')
+#   $with_facts = run_plan(facts::external, $targets, path => file::join($moduledir, 'facts.d'))
+#   return $with_facts.map |$target| { $target.facts }
+plan facts::external(
+  String $path,
+  TargetSpec $targets
+) {
+  $t = get_targets($targets)
+
+  # Strip out dotfiles
+  $facts_paths = dir::children($path).filter |$path| { $path[0] != '.' }
+  $facts_paths.each |$file| {
+    $ext_facts = run_script(file::join($path, $file), $t)
+    $t.each |$target| {
+      $raw_facts = $ext_facts.find($target.name).value['stdout']
+      $jsonoryaml = catch_errors() || {
+        parseyaml($raw_facts)
```

Comment:
`parseyaml` is part of the `stdlib` module, which Bolt does not currently ship with. We should make sure to ship with `stdlib` before shipping this plan with Bolt.

Reply:
https://github.com/puppetlabs/bolt/blob/main/Puppetfile#L31 We do ship with it.

Reply:
Haha, whoops. I guess I deleted that from my Puppetfile at some point. 😅